### ISSUE TITLE
Enable macros for scripts

### ISSUE THREAD

**Issue by LeXXik** [playcanvas/editor#451]
It would be nice to be able to have a macro that the scripts could use to filter out some debugging parts. Similar to the engine side:
```js
// #if _DEBUG
        
console.log('some debug info')
        
// #endif
```

Perhaps an ability to define the macros for build vs dev would be nice.

**Comment by yaustar** [comment]
(As a side note, some developers have done a post processing phase using the REST API without minification, and doing the strip out and minification in their own processing script)

**Comment by Maksims** [comment]
Simplest thing would be:
When engine compiled with `DEBUG` define, set `pc.debug` property to true.
So in your scripts you then could do:
```js
if (pc.debug) {
    console.log('some logs');
}
```

**Comment by yaustar** [comment]
That will keep the code in the published build though

**Comment by Maksims** [comment]
> That will keep the code in the published build though unless you thinking of stripping it out on publishing too?
Published builds can use debug version when building to staging servers for example. Also, it provides an ability to set `pc.debug` to `true` when in production, that way enabling logging. So it is totally a valid path.

**Comment by LeXXik** [comment]
Hmm, I thought `pc.debug` is used for something else, like output some data for debugging, e.g.:
```js
pc.debug.display({ a: 1 });
```
![image](https://user-images.githubusercontent.com/5677782/124897043-97f7be80-dfe6-11eb-9bc4-3b0d05345405.png)

And yes, I am referring to builds hosted and launched on PC servers. For those that download the build to process it later in CD/CI is not an issue, as it handles it well there.

**Comment by Maksims** [comment]
If `pc.debug` is taken, use another property name.
There are tons of options to enable debugging, like check for a query parameter in URL, or host URL, etc. Different teams find many options.

**Comment by LeXXik** [comment]
Yes, I agree. This is a a nice to have feature, rather than something critical. We do it differently in different projects and that is my point - every team uses own way, which you either have to know about or develop. It would be nice to have a feature that everyone is using anyway to be a part of the IDE.
